flink-scala/src/main/scala/org/apache/flink/api/scala/GroupedDataSet.scala

```diff
@@ -282,27 +283,57 @@ class GroupedDataSet[T: ClassTag](
   }
 
   /**
-   * Creates a new [[DataSet]] by merging the elements of each group (elements with the same key)
-   * using an associative reduce function.
-   */
+    * Creates a new [[DataSet]] by merging the elements of each group (elements with the same key)
+    * using an associative reduce function.
+    */
   def reduce(fun: (T, T) => T): DataSet[T] = {
+    reduce(getCallLocationName(), fun, CombineHint.OPTIMIZER_CHOOSES)
+  }
+
+  /**
+   * Special [[reduce]] operation for explicitly telling the system what strategy to use for the
+   * combine phase.
+   * If null is given as the strategy, then the optimizer will pick the strategy.
+   */
+  def reduce(fun: (T, T) => T, strategy: CombineHint): DataSet[T] = {
```

Comment:
> Would it be OK if I added optional arguments instead of the overloads like this?

Oops, this doesn't work. I'm getting
```
Error:(45, 7) in class GroupedDataSet, multiple overloaded alternatives of reduce define default arguments
```

An alternative solution would be to instead of just wrapping in a Scala `DataSet`, create a Scala class that wraps `ReduceOperator`, has the same `setCombineHint` method, and inherits from the Scala `DataSet`.